src/app/containers/RadioSchedule/Canonical/index.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState, useContext } from 'react';
+import 'isomorphic-fetch';
+import { string } from 'prop-types';
+import styled from 'styled-components';
+import SectionLabel from '@bbc/psammead-section-label';
+import { Link } from '@bbc/psammead-story-promo';
+import { ServiceContext } from '#contexts/ServiceContext';
+import webLogger from '#lib/logger.web';
+
+const logger = webLogger();
+
+const CanonicalRadioSchedule = ({ endpoint }) => {
+  const header = 'BBC News Radio';
+  const [schedule, setRadioSchedule] = useState([]);
+  const {
+    service,
+    script,
+    dir,
+    radioSchedule: { frequenciesPageUrl },
+  } = useContext(ServiceContext);
+
+  const handleResponse = async response => {
+    const radioScheduleData = await response.json();
+    setRadioSchedule(radioScheduleData.schedules.slice(0, 4));
+  };
+
+  useEffect(() => {
+    const fetchRadioScheduleData = pathname =>
+      fetch(pathname, { mode: 'no-cors' })
+        .then(handleResponse)
+        .catch(e => logger.error(`HTTP Error: "${e}"`));
+
+    fetchRadioScheduleData(endpoint);
+  }, [endpoint]);
+
+  const RadioFrequencyLink = styled(Link)`
```

Comment:
Could you create these components outside of the main CanonicalRadioSchedule component?

Reply:
Moved `RadioFrequencyLink` above CanonicalRadioSchedule